packages/relay/src/lib/services/ethService/contractService/ContractService.ts

```diff
@@ -255,11 +255,7 @@ export class ContractService implements IContractService {
           return CommonService.redirectBytecodeAddressReplace(address);
         } else if (result.type === constants.TYPE_CONTRACT) {
           if (result.entity.runtime_bytecode !== constants.EMPTY_HEX) {
-            const prohibitedOpcodes = ['CALLCODE', 'DELEGATECALL', 'SELFDESTRUCT', 'SUICIDE'];
-            const opcodes = disassemble(result.entity.runtime_bytecode);
-            const hasProhibitedOpcode =
-              opcodes.filter((opcode) => prohibitedOpcodes.indexOf(opcode.opcode.mnemonic) > -1).length > 0;
-            if (!hasProhibitedOpcode) {
+            if (blockNumber !== 'latest') {
```

Comment:
Does it make sense to move the cache set right after the `HTS_ADDRESS` comparison above?

Skimming through the logs in testnet, around 50% of `eth_getCode` calls pass an HTS token address. The main reason to cache HTS bytecode calls is to avoid calls to `resolveEntityType` and `getHistoricalBlockResponse`.

However, both `resolveEntityType` and `getHistoricalBlockResponse` are cached as well. That means we save only 1 cache request in the best scenario (when all required data is cached).